doc/testing.md

```diff
@@ -4,8 +4,94 @@
 
 If you are unit testing your application you should stub all external services.
 
+### Interpreter
+
+If you are using tapir do describe your API and run using some server interpreter then you might want to test your
+endpoints definitions (like validations, decoding) as well as your custom server options (like interceptors, failure handlers) while 
+stubbing the actual response of your server logic.
+
+Add the dependency:
+
+```scala
+"com.softwaremill.sttp.tapir" %% "tapir-sttp-stub-server" % "@VERSION@"
+```
+
+Let's assume you are using Akka interpreter, then given the following endpoint:
+
+```scala mdoc:invisible
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import sttp.client3._
+import sttp.model.StatusCode
+import sttp.monad.FutureMonad
+import sttp.tapir._
+import sttp.tapir.server.ServerEndpoint.Full
+import sttp.tapir.server.akkahttp.AkkaHttpServerOptions
+import sttp.tapir.server.interceptor.exception.ExceptionContext
+import sttp.tapir.server.interceptor.{CustomInterceptors, ValuedEndpointOutput}
+import sttp.tapir.server.stub.TapirStubInterpreter
+import scala.concurrent.Future
+```
+
+```scala mdoc:silent
+val greetUser: Full[String, String, Unit, String, String, Any, Future] = endpoint.get
```

Comment:
do we need the full type annotation here? Wouldn't it be sufficient to use `ServerEndpoint[Any,Future]`?

Reply:
If someone want's to stub server logic then `Full` definition has to be provided since `thenSuccess/thenError` needs `O`, `E` types.

Reply:
right, but I suppose for simple examples you can just use the simpler type?

Reply:
sure